zebra-chain/src/addresses.rs

```diff
@@ -0,0 +1,252 @@
+    /// bytes. These are big-endian.
+    /// https://en.bitcoin.it/Base58Check_encoding#Encoding_a_Bitcoin_address
+    fn hash_payload(bytes: &[u8]) -> [u8; 20] {
+        let sha_hash = Sha256::digest(bytes);
+        let ripe_hash = Ripemd160::digest(&sha_hash);
+        let mut payload = [0u8; 20];
+        payload[..].copy_from_slice(&ripe_hash[..]);
+        payload
+    }
+}
+
+impl From<Script> for TransparentAddress {
+    fn from(script: Script) -> Self {
+        TransparentAddress::PayToScriptHash {
+            network: Network::Mainnet,
+            script_hash: Self::hash_payload(&script.0[..]),
+        }
+    }
+}
+
+impl From<PublicKey> for TransparentAddress {
+    fn from(pub_key: PublicKey) -> Self {
+        TransparentAddress::PayToPublicKeyHash {
+            network: Network::Mainnet,
+            pub_key_hash: Self::hash_payload(&pub_key.serialize()[..]),
+        }
+    }
+}
+
+impl fmt::Debug for TransparentAddress {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let mut bytes = io::Cursor::new(Vec::new());
+        let _ = self.zcash_serialize(&mut bytes);
+
+        f.debug_tuple("TransparentAddress")
+            .field(&bs58::encode(bytes.get_ref()).with_check().into_string())
+            .finish()
+    }
+}
+
+impl ZcashSerialize for TransparentAddress {
+    fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), io::Error> {
+        match self {
+            TransparentAddress::PayToScriptHash {
+                network,
+                script_hash,
+            } => {
+                // Dev network doesn't have a recommendation so we
+                // default to testnet bytes if it's not mainnet.
+                if *network == Network::Mainnet {
+                    writer.write_all(&[0x1C, 0xBD][..])?
+                } else {
+                    writer.write_all(&[0x1C, 0xBA][..])?
+                }
+                writer.write_all(script_hash)?
+            }
+            TransparentAddress::PayToPublicKeyHash {
+                network,
+                pub_key_hash,
+            } => {
+                // Dev network doesn't have a recommendation so we
+                // default to testnet bytes if it's not mainnet.
+                if *network == Network::Mainnet {
+                    writer.write_all(&[0x1C, 0xB8][..])?
+                } else {
+                    writer.write_all(&[0x1D, 0x25][..])?
+                }
+                writer.write_all(pub_key_hash)?
+            }
+        }
+
+        Ok(())
+    }
+}
+
+impl ZcashDeserialize for TransparentAddress {
+    fn zcash_deserialize<R: io::Read>(mut reader: R) -> Result<Self, SerializationError> {
+        let mut version_bytes = [0; 2];
+        reader.read_exact(&mut version_bytes)?;
+
+        let mut hash_bytes = [0; 20];
+        reader.read_exact(&mut hash_bytes)?;
+
+        match version_bytes {
+            [0x1c, 0xbd] => Ok(TransparentAddress::PayToScriptHash {
```

Comment:
Adding named constants for this would avoid the repetition of magic values between encoding and decoding, and allow them to be reviewed in one place.